### PR TITLE
Update doc examples feature extractor -> image processor

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Replaces vision feature extractor references with image processors throughout transformers documentation. 

Places where changes didn't happen: 
* Vision models which use a `Processor` class. Some processor classes still have `feature_extractor_class` property, to be removed in future. 
* `examples/...` - required changes to code outside to scope of this PR and dependant on some changes to `Processor` class

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/source/en/model_doc/cvt.mdx [hunk 1]
```diff
@@ -32,7 +32,7 @@ a crucial component in existing Vision Transformers, can be safely removed in ou
 Tips:
 
 - CvT models are regular Vision Transformers, but trained with convolutions. They outperform the [original model (ViT)](vit) when fine-tuned on ImageNet-1K and CIFAR-100.
-- You can check out demo notebooks regarding inference as well as fine-tuning on custom data [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/VisionTransformer) (you can just replace [`ViTFeatureExtractor`] by [`AutoFeatureExtractor`] and [`ViTForImageClassification`] by [`CvtForImageClassification`]).
+- You can check out demo notebooks regarding inference as well as fine-tuning on custom data [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/VisionTransformer) (you can just replace [`ViTFeatureExtractor`] by [`AutoImageProcessor`] and [`ViTForImageClassification`] by [`CvtForImageClassification`]).
```
Comment (ydshieh) [inline docs/source/en/model_doc/cvt.mdx:35]: Should we change `ViTFeatureExtractor` to `ViTImageProcessor` here?

File: docs/source/en/tasks/image_classification.mdx [hunk 2]
```diff
@@ -213,7 +213,7 @@ At this point, only three steps remain:
 ...     data_collator=data_collator,
 ...     train_dataset=food["train"],
 ...     eval_dataset=food["test"],
-...     tokenizer=feature_extractor,
+...     tokenizer=image_processor,
```
Comment (ydshieh) [inline docs/source/en/tasks/image_classification.mdx:216]: I don't mean to change this in current PR - just feel mind blowing when I see this line 😲 😲 😲 😲

File: docs/source/en/autoclass_tutorial.mdx [hunk 3]
```diff
@@ -23,6 +23,7 @@ Remember, architecture refers to the skeleton of the model and checkpoints are t
 In this tutorial, learn to:
 
 * Load a pretrained tokenizer.
+* Load an image processor
```
Comment (ydshieh) [inline docs/source/en/autoclass_tutorial.mdx:26]: Guess you deliberately removed `pretrained`? No strong opinion, just not super sure if we prefer to keep it as the same as with other entries here.

File: docs/source/en/task_summary.mdx [hunk 4]
```diff
@@ -1103,22 +1103,22 @@ Class tiger cat with score 0.0229
 
 The general process for using a model and feature extractor for image classification is:
 
-1. Instantiate a feature extractor and a model from the checkpoint name.
-2. Process the image to be classified with a feature extractor.
+1. Instantiate an image processor and a model from the checkpoint name.
```
Comment (ydshieh) [inline docs/source/en/task_summary.mdx:1106]: One line above should be

```
The general process for using a model and an image processor for image classification is:
```

**Review by ydshieh — APPROVED** [review]
LGTM, thank you @amyeroberts this is huge and impactful work!

Left a few comments.

File: docs/source/en/model_doc/cvt.mdx (hunk 1 above)
Comment (amyeroberts) [reply docs/source/en/model_doc/cvt.mdx:35]: Good question. I wasn't sure as the notebook still has `VitFeatureExtractor` - is it more confusing not being consistent  here or instructing to change on an object that's not there?

File: docs/source/en/tasks/image_classification.mdx (hunk 2 above)
Comment (amyeroberts) [reply docs/source/en/tasks/image_classification.mdx:216]: I know 👀  - I found this pretty confusing tbh when I first saw it 🙈

File: docs/source/en/autoclass_tutorial.mdx (hunk 3 above)
Comment (amyeroberts) [reply docs/source/en/autoclass_tutorial.mdx:26]: Nope. I'll add it back. No image processors are 'pretrained' - but they are preconfigured and it's possible that there's some learnt element added to them in the future.

File: docs/source/en/task_summary.mdx (hunk 4 above)
Comment (amyeroberts) [reply docs/source/en/task_summary.mdx:1106]: Yep, good catch.

File: docs/source/en/autoclass_tutorial.mdx (hunk 3 above)
Comment (amyeroberts) [inline docs/source/en/autoclass_tutorial.mdx:26]:
```suggestion
* Load a pretrained image processor
```

File: docs/source/en/task_summary.mdx [hunk 5]
```diff
@@ -1103,22 +1103,22 @@ Class tiger cat with score 0.0229
 
 The general process for using a model and feature extractor for image classification is:
```
Comment (amyeroberts) [inline docs/source/en/task_summary.mdx:1104]:
```suggestion
The general process for using a model and image processor for image classification is:
```

File: docs/source/en/model_doc/cvt.mdx (hunk 1 above)
Comment (ydshieh) [reply docs/source/en/model_doc/cvt.mdx:35]: OK, I read the paragraph too fast. Keep as it is is fine :-)

**Review by sgugger — APPROVED** [review]
Amazing work, thanks a lot!

**Pull request merged**